### DISCUSSION TITLE
Add js source files back to tagged releases.

### DISCUSSION THREAD

**Discussion by myleshyson** [craftcms/cms#12050] [Ideas]
I'd like to request the source js files for Craft be added back instead of being removed. Garnish is really the only way to _fully_ extend the control panel. So for any developer, plugin or otherwise, that needs to extend any core garnish object, there's not a easy way anymore to get ide completion/search on any of those objects. 

It would be one thing if there was documentation for Garnish. There's not though, so code diving is the only way to see what can be extended and how. 

My current workflow is to open up the craft repo in github1s and scroll/search until I maybe find what I'm looking for.

I've avoided pulling in the craft cms source and building it since it's kind of an annoying workflow. Ideally I can keep updating craft for our site like normal _and_ still have access to source files for extending various core js objects. 

I know it's a little bit of a weird ask but I'm sure there's a lot of devs that would appreciate some ide help with those files as well.

**Comment by brandonkelly** [comment]
> My current workflow is to open up the craft repo in github1s and scroll/search until I maybe find what I'm looking for.

Alternatively, you could pull the Craft repo into `vendor/`, either with [composer-link](https://github.com/SanderSander/composer-link) or by adding this to composer.json:

```json
"repositories": [
  {
    "type": "git", 
    "url": "https://github.com/craftcms/cms.git"
  }
]
```
